extensions/whox.md

```diff
@@ -0,0 +1,83 @@
+---
+title: WHOX
+layout: spec
+copyrights:
+  -
+    name: "Simon Ser"
+    period: "2021"
+    email: "contact@emersion.fr"
+---
+
+## Introduction
+
+The `WHO` command allows clients to obtain information about other clients.
+However, `WHO` doesn't allow clients to request extra data (e.g. account names)
+nor a subset of the data (e.g. omit user and host).
+
+## `ISUPPORT` token
+
+Servers supporting this specification MUST include the `WHOX` token in
+`RPL_ISUPPORT` without any value. Clients MUST accept `WHOX` tokens with a
+value for forwards compatibility.
+
+## Extended `WHO` command
+
+The `WHO` command is extended with an optional argument:
+
+    WHO <mask> [%<fields>[,<token>]]
+
+The second argument contains a percent character followed by a list of fields
+requested by the client. Each field is represented by a letter. The standard
+fields are:
+
+- 't': return the `<token>` specified by the client
+- 'c': return an arbitrary channel the client is joined to
+- 'u': return the username
+- 'i': return the IP address
+- 'h': return the hostname
+- 's': return the server name
+- 'n': return the nickname
+- 'f': return the `WHO` flags (away, server operator, etc)
+- 'l': return the number of seconds the user has been idle for
+- 'a': return the account name
+- 'r': return the realname
+
+Servers MAY support additional non-standard fields.
+
+Clients can also specify a token which will be returned by the server in the
+replies. The token MUST contain only digit characters and MUST contain at most
+3 characters. Clients MUST NOT include 't' in `<fields>` without specifying a
+token.
+
+Servers MAY support additional non-standard flags before the percent character.
+
+The server will reply with zero, one or more `RPL_WHOSPCRPL` replies, followed
+by a final `RPL_ENDOFWHO` reply. Servers MUST NOT send `RPL_WHOREPLY` replies.
+
+## `RPL_WHOSPCRPL` (354) numeric reply
+
+    :<server> 354 <client> [token] [channel] [user] [ip] [host] [server] [nick] [flags] [idle] [account] [:realname]
```

Comment:
AIUI the ordering of the fields sent by the client MUST be ignored, and then the server MUST return the requested fields in the order specified here? If so, it might be worth stating this explicitly, and possibly also adding an example where the client requests in a nonstandard order and the server returns in the standard order.

Reply:
> 

This is correct as per both my understanding and the implementations I have observed

Reply:
Done!